### ISSUE TITLE
Is this project dead?

### ISSUE THREAD

**Issue by famda** [dotnet-state-machine/stateless#466]

**Comment by Allann** [comment]
wondering that too.

**Comment by Blackclaws** [comment]
I think its relatively feature complete and there isn't that much room for improvement anymore, there are a couple of issues remaining and open pull requests and those might be interesting to still solve but I think it seems less dead and more "done"

**Comment by NelloRizzo** [comment]
no... but i not have more collaborators... if you intend to continuate, i'm ok!

**Comment by ffMathy** [comment]
@NelloRizzo but there are plenty of pull requests pending merge. Do you intend to review these and get them merged?

**Comment by HenningNT** [comment]
I do not have the capacity to do anything anymore.
If someone wants to take it over, please do.

**Comment by ffMathy** [comment]
But that's literally what people have done. A lot of pull requests are already up.

**Comment by HenningNT** [comment]
Managing a project is more than accepting PRs.

**Comment by HenningNT** [comment]
@ffMathy Would you like a go?

**Comment by mclift** [comment]
As a past contributor to stateless, I'd consider joining the project if you're stuck, @HenningNT.

**Comment by ffMathy** [comment]
Yeah I'd like to become an official contributor! Feel free to add me.

**Comment by HenningNT** [comment]
Thank you!

**Comment by HenningNT** [comment]
As @nblumhardt is the original owner of this project he also has the Nuget key, so he will need to be involved at some point to release new versions.

**Comment by nblumhardt** [comment]
Thanks for the mention @HenningNT . Stateless is an organisation on NuGet.org, so you can also access/update/modify keys if you wish - just in case you ever need to do this.

I don't think it's worthwhile for this project to take new maintainers on board without one of the longer-term maintainers to help keep things heading in a consistent direction. Since you (understandably) and I are not in a position to do that, I think we should consider keeping this project in "done" mode (emergency/critical fixes only), and help those who are interested in carrying it forward to create a fork that we can link to.

 I think this is sensible because:

 * If the new maintainers want to do significant work on the project/add features/maintain it long-term, then the initial effort of creating the fork, renaming the package, and adding some links to it here in the README is reasonable;
 * If the new maintainers just need a patch or two and don't end up being involved long-term, then passing over the reins now will just leave this project in exactly the same state (LOL) a short while down the track (while still requiring non-trivial handover efforts)

Does this line of reasoning make sense? And since the current project has such a terrible name - what would the "blessed" fork be called? :-)

**Comment by vslee** [comment]
Good points @nblumhardt. Perhaps I can argue the opposite though :)
I think we should avoid forking the project for the following reasons:

- This project already has built up a following, including issues/stars/watchers on GitHub, which would not transfer to the new fork. 
- There would be confusion as to which is the blessed fork. We can put a link the README like you said, but that would not be easily visible in NuGet (especially when searching for packages in Visual Studio). The new project may not even show up in a search for `stateless` on NuGet if the name is changed.
- There'd be the question of whether to "archive" this repo, in which case no more issues or PRs would be able to be created. This would mean no emergency/critical fixes either. The other option would be to leave it open. But then the community would be split, with some people still opening issues/PRs on this repo instead of the new fork. 

One solution to the issues you pointed out above, without forking, would be to move to the .NET Foundation. This way they would be able to hold the reins long term, while allowing our maintainers the freedom handle day to day tasks. If the new maintainers stick around for the long term, then great. But if they are short term like you mentioned, then the foundation can still hold the reins long term. There are also other [benefits](https://dotnetfoundation.org/projects/why-join), and we would be in the company of other [big and small projects](https://dotnetfoundation.org/projects) who have already joined.

Acceptance into the .NET Foundation is not guaranteed and would require some work to meet their [eligibility requirements](https://dotnetfoundation.org/projects/net-foundation-project-committee#eligibility-criteria). I would be happy to do that work, but only if this is what the top contributors (@nblumhardt and @HenningNT) want.

**Comment by crozone** [comment]
I'm also happy to throw my hat in to be a maintainer. I use this library quite extensively so it'd be well worth my time, even just as an extra reviewer of PRs, I already have a few nuget packages of my own so I know the challenges involved.

@nblumhardt  I don't think that forking the project is the best way forward - most of the PRs appear to be for bugfixes rather than large features, which is consistent with the idea that the project is feature complete. Meanwhile a new fork implies the project is heading off in some other direction, but there doesn't appear to be an obvious new direction for it to go.

@vslee .NET Foundation would be a huge step and incredibly valuable IMHO. This project is more or less *the* formal state machine library for .NET and I hope that is sustained into the future.

**Comment by nblumhardt** [comment]
Thanks @vslee and @crozone for replying! I think my reply above was badly timed, I didn't notice that things have already gone ahead, so perhaps there's not a lot of point digging into it further :-)

The main challenge for me personally is that I haven't been deeply involved with the project for quite a while, again. I don't have a great picture of who is actively contributing at what level, and because I don't have the time to work alongside new contributors, I have no way of ensuring the project stays aligned with the needs of its current users.

I think @HenningNT has a much better idea of this than me, though, and I'm happy to defer to him entirely on these decisions. It's probably time for me to remove myself from the maintainer team for this project, as I can't really hold up my end of the bargain, there.

I do still think this is a great library that deserves to live on, though! Over the years it's grown and matured nicely thanks to a lot of contributions and feedback - I agree that it would be a shame to put any stumbling blocks in front of it. I'm just not in a position to help drive it anymore, unfortunately.

**Comment by HenningNT** [comment]
I'm very happy that someone is taking over this project! I think we should keep the name, package and everything, so that any confusion is avoided. I've may be been  too cautious about upsetting users, so I haven't changed much.

You should do what you want with this now.

Good luck!

**Comment by HenningNT** [comment]
About the  .NET Foundation thing... I have no idea what that would mean for this project. If you think it's a good thing then I trust you know what you are doing :-)

**Comment by HenningNT** [comment]
@nblumhardt I think you need to update the nuget API key.
https://github.com/dotnet-state-machine/stateless/runs/8186991968?check_suite_focus=true#step:7:30

**Comment by nblumhardt** [comment]
Due to time constraints as mentioned above, I'm leaving administration of this project. Stateless is an organization on both GitHub and NuGet, currently @HenningNT and I are the only admins in each.

@HenningNT to keep this project rolling without me/us, I'll add @ffMathy and @mclift as organization owners both here and on NuGet.

Folks - can you please let me know your NuGet usernames so that I can add you there? Thanks!

**Comment by mclift** [comment]
@nblumhardt oops, missed your question! I'm mclift on NuGet

**Comment by nblumhardt** [comment]
Done! Thanks @mclift 👍

**Comment by ffMathy** [comment]
Mine is ffMathy.

**Comment by mclift** [comment]
Thanks, @nblumhardt 👍  NuGet API key updated.

**Comment by crozone** [comment]
Awesome, 5.11.0-dev-3191463566 is live on NuGet. We probably just need a version bump now?

**Comment by mclift** [comment]
Sure, I'll prepare a changelog and PR for a version bump when I get a chance, unless someone else gets there first 😄

Should we close this issue now btw, and move general chat to the Discussions tab?

**Comment by nblumhardt** [comment]
Done! Just one task to go before I think we can close this - I've sent invitations to @ffMathy and @mclift to join the organization; once they're accepted I'll leave the org myself 🥲 and close this.

**Comment by vslee** [comment]
> I've sent invitations to @ffMathy and @mclift to join the organization; once they're accepted I'll leave the org myself 🥲 and close this.

Thank you for passing the baton, but please don't leave the org yourself. If we decide to join the .net foundation, they will need the top contributors to sign a CLA. They define those as "typically the top 2-5 developers who have contributed more than a few hundred lines of code." While I'm sure that the foundation can figure out that you're a top contributor even if you leave the org, it might be clearer if you don't leave. 
Also, just don't want the project to ever fall into an orphan status, where there are no longer any active admins, and no one to assign new ones. (you could always "unwatch" this repo if there are too many notifications)

**Comment by nblumhardt** [comment]
Appreciate the sentiment @vslee!

There should soon be enough admins to ensure there will always be some path forwards.

Things will work out better in the end if we put our trust in the new maintainers - I don't scale well, so although me being here might seem like a backup option, my attention and time are stretched very very thinly, so there are others who'll do a better job if I get out of the way :-)

**Comment by mclift** [comment]
Hey @crozone, did you get a maintainer invitation?

**Comment by crozone** [comment]
@mclift No, not yet I don't think. I'm still interested though :)

**Comment by mclift** [comment]
@crozone sweet :) You should be invited now

**Comment by mclift** [comment]
That said, I think it might need someone with organisation admin access to add you properly. Have I done it wrong, @nblumhardt / @HenningNT?

**Comment by crozone** [comment]
Cheers, I seem to be an admin now! It looks like @p-m-j is the only member of the parent organization, is that expected?

**Comment by p-m-j** [comment]
@crozone the only public member, there are 7 private members of the org

**Comment by mclift** [comment]
I just took another look at the org. nblumhardt and @JeremyCade are the current owners.

**Comment by nblumhardt** [comment]
I've made you an owner now @mclift; if you're able to see org settings etc., would you then be so kind as to remove me from the org? Thanks for carrying things forward, I still love this little project so it's awesome that it's able to survive my own unfortunately lack of time :-)

**Comment by mclift** [comment]
We may put out a call for new maintainers soon. In the meantime, I think this issue can be closed.

To answer the question: the project isn't dead, there are just times when the maintainers are busy with other projects.
